src/main/java/org/influxdb/impl/InfluxDBImpl.java

```diff
@@ -187,18 +188,20 @@ public InfluxDBImpl(final String url, final String username, final String passwo
     setRetentionPolicy(retentionPolicy);
   }
 
-  private InetAddress parseHostAddress(final String url) {
-      HttpUrl httpUrl = HttpUrl.parse(url);
+  private String parseHost(final String url) {
+    HttpUrl httpUrl = HttpUrl.parse(url);
```

Comment:
Is `HttpUrl` able to parse URIs that do not start with the string `http`? IMO we should treat this parameter as an `URI` and not an `URL`. The main reasons are:
* Users may finally stop using the scheme "http://" to connect via UDP. [RFC 1738 for URL](https://tools.ietf.org/html/rfc1738#section-3) does not allow the scheme "udp://" but the [RFC 3986 for URI](https://tools.ietf.org/html/rfc3986#section-3.1) does.
* If so, this line could be replaced with `URI.create(...)` and later you could use `URI#getHost()` to get only the hostname.